### ISSUE TITLE
failed to read thrift

### ISSUE THREAD

**Issue by jaymzh** [dequis/purple-facebook#357]
I'm getting:

```
Failed to read thrift: api.c:1679 fb_api_cb_publish_pt: assertion 'FALSE' failed
```

when I try to connect.

Version: 0.9.4-c9b74a765767

**Comment by dequis** [comment]
https://github.com/dequis/purple-facebook/wiki/Debugging-Logs

**Comment by jaymzh** [comment]
Yup. Logs sent.

**Comment by dequis** [comment]
Hmmm, odd thing, there's a sixth field in the presence update struct but even the latest released beta version of the android app doesn't have it.

Well. Thrift is supposed to be designed to handle this sort of thing gracefully - we really should be ignoring unknown fields already. And the way this code is written, it can't do that.

Here, have a patch:

https://github.com/dequis/purple-facebook/tree/mystery-meat-2

**Comment by jaymzh** [comment]
Can confirm this works, thanks!

**Comment by dequis** [comment]
Whoops sorry reopened accidentally. Anyway this is fixed in 0.9.5

https://github.com/dequis/purple-facebook/releases/tag/v0.9.5-9ff9acf9fa14

**Comment by bchgys** [comment]
Thank-you for the quick fix. Can confirm this solved the 'Failed to read thrift' issue for pidgin/libpurple 2.12.0  on win7_x64_sp1. Much appreciated. G
